src/sinks/util/encoding/mod.rs

```diff
@@ -106,12 +112,84 @@ pub trait Encoder<T> {
     }
 }
 
-impl<E, T> Encoder<T> for Arc<E>
-where
-    E: Encoder<T>,
-{
-    fn encode_input(&self, input: T, writer: &mut dyn io::Write) -> io::Result<usize> {
-        (**self).encode_input(input, writer)
+#[cfg(feature = "codecs")]
+impl Encoder<Vec<Event>> for (Transformer, crate::codecs::Encoder<()>) {
+    fn encode_input(&self, events: Vec<Event>, writer: &mut dyn io::Write) -> io::Result<usize> {
+        let mut encoder = self.1.clone();
+        let mut bytes_written = match encoder.serializer() {
+            codecs::encoding::Serializer::Json(_) => writer.write(b"[")?,
+            codecs::encoding::Serializer::RawMessage(_) => 0,
+        };
+
+        bytes_written += events
+            .into_iter()
+            .map(|mut event| {
+                self.0.transform(&mut event);
+                let mut bytes = BytesMut::new();
+                encoder
+                    .encode(event, &mut bytes)
+                    .map_err(|error| io::Error::new(io::ErrorKind::InvalidData, error))?;
+                Ok(writer.write(&bytes)?
+                    + match encoder.serializer() {
+                        codecs::encoding::Serializer::Json(_) => writer.write(b",")?,
```

Comment:
It looks like this will always produce a `,` at the end, which is not allowed in JSON.